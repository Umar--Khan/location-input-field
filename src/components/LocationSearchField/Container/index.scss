.location-page-container {
  background-color: $color-blue;
  min-height: 70vh;
}

.location-select-container {
  background-color: $color-yellow;
  margin: 0 $space-060;
  padding: $space-060;
  display: flex;
  flex-direction: column;
}

.title {
  color: $color-white;
  text-align: center;
  font-size: $font-060;
}
