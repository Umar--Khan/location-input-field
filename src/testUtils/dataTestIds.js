export const page = {
  pageTitle: 'page-title',
  locationSearchFieldTitle: 'location-search-field-title',
};

export const tags = {
  tagLabel: 'tags-label',
};

export const suggestions = {
  suggestionItem: 'suggestion-item',
  suggestionItemCountry: 'suggestion-item-country',
  suggestionItemName: 'suggestion-item-name',
  container: 'suggestions-container',
};

export const searchField = {
  inputField: 'search-input-field',
  inputFieldLabel: 'search-input-field-label',
};

export const app = {
  appContainer: 'app-container',
};
