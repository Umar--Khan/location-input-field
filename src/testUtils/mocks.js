export const locationSuggestions = [
  {
    'country': 'Bulgaria',
    'lng': 23.7,
    'searchType': 'G',
    'alternative': ['BG'],
    'index': 1,
    'bookingId': 'city-355866',
    'placeType': 'C',
    'placeKey': '100948',
    'countryIso': 'bg',
    'locationId': '-1',
    'name': 'Kʼoseva',
    'ufi': -835331,
    'isPopular': false,
    'region': 'Sofia',
    'lang': 'en',
    'lat': 42.4,
  },
  {
    'country': 'Bulgaria',
    'lng': 25.15,
    'searchType': 'G',
    'alternative': ['BG'],
    'index': 2,
    'bookingId': 'city-355411',
    'placeType': 'C',
    'placeKey': '100946',
    'countryIso': 'bg',
    'locationId': '-1',
    'name': 'Kʼochevtsi',
    'ufi': -835200,
    'isPopular': false,
    'region': 'Kardzhali',
    'lang': 'en',
    'lat': 41.6833,
  },
  {
    'country': 'Bulgaria',
    'lng': 25.3,
    'searchType': 'G',
    'alternative': ['BG'],
    'index': 3,
    'bookingId': 'city-355867',
    'placeType': 'C',
    'placeKey': '100949',
    'countryIso': 'bg',
    'locationId': '-1',
    'name': 'Kʼosevo',
    'ufi': -835333,
    'isPopular': false,
    'region': 'Kardzhali',
    'lang': 'en',
    'lat': 41.6167,
  },
  {
    'country': 'Bulgaria',
    'lng': 26.25,
    'searchType': 'G',
    'alternative': ['BG'],
    'index': 4,
    'bookingId': 'city-355869',
    'placeType': 'C',
    'placeKey': '100950',
    'countryIso': 'bg',
    'locationId': '-1',
    'name': 'Kʼosevtsi',
    'ufi': -835335,
    'isPopular': false,
    'region': 'Targovishte',
    'lang': 'en',
    'lat': 43.1,
  },
  {
    'country': 'Bulgaria',
    'lng': 26.6833,
    'searchType': 'G',
    'alternative': ['BG'],
    'index': 5,
    'bookingId': 'city-355624',
    'placeType': 'C',
    'placeKey': '100947',
    'countryIso': 'bg',
    'locationId': '-1',
    'name': 'Kʼolmen',
    'ufi': -835239,
    'isPopular': false,
    'region': 'Shumen',
    'lang': 'en',
    'lat': 43.0333,
  },
  {
    'country': 'Georgia',
    'lng': 41.4436,
    'searchType': 'G',
    'alternative': ['GE'],
    'index': 6,
    'bookingId': 'city-876445',
    'placeType': 'C',
    'placeKey': '449544',
    'countryIso': 'ge',
    'locationId': '-1',
    'name': 'Kʼochʼara',
    'ufi': -2328015,
    'isPopular': false,
    'lang': 'en',
    'lat': 42.8575,
  },
];
