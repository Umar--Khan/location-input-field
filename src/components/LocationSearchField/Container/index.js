import { page } from 'testUtils/dataTestIds';

const Container = ({ children }) => (
  <main className='location-page-container'>
    <div className='location-page-container__wrapper'>
      <h1
        className='location-page-container__title'
        data-testid={page.pageTitle}
      >
        Car Hire - Search, Compare &#38; Save
      </h1>
      <div className='location-select-container'>
        <h2
          className='location-select-container__title'
          data-testid={page.locationSearchFieldTitle}
        >
          Lets find your ideal car
        </h2>
        {children}
      </div>
    </div>
  </main>
);

export default Container;
