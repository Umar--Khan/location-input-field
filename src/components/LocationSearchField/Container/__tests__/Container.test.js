import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import Container from 'components/LocationSearchField/Container';
import { page } from 'testUtils/dataTestIds';

const component = <Container>Content</Container>;

describe('<Container />', () => {
  it('Render snapshot correctly', () => {
    const component = renderer.create(<Container />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Render correct page title', () => {
    const { getByTestId } = render(component);

    expect(getByTestId(page.pageTitle).textContent).toBe(
      'Car Hire - Search, Compare & Save',
    );
  });

  it('Render correct location search field title', () => {
    const { getByTestId } = render(component);

    expect(getByTestId(page.locationSearchFieldTitle).textContent).toBe(
      'Lets find your ideal car',
    );
  });

  it('Render children', () => {
    const { getByText } = render(component);

    expect(getByText('Content')).toBeInTheDocument();
  });
});
