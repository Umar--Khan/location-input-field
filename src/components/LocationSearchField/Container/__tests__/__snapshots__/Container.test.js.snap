// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Render snapshot correctly 1`] = `
<main
  className="location-page-container"
>
  <div
    className="location-page-container__wrapper"
  >
    <h1
      className="location-page-container__title"
      data-testid="page-title"
    >
      Car Hire - Search, Compare & Save
    </h1>
    <div
      className="location-select-container"
    >
      <h2
        className="location-select-container__title"
        data-testid="location-search-field-title"
      >
        Lets find your ideal car
      </h2>
    </div>
  </div>
</main>
`;
